UnityMirror/UnityServer/Assets/Server/Scripts/UnityNetworkServer.cs

```diff
@@ -57,29 +57,32 @@ private void OnReceiveAuthenticate(NetworkMessage netMsg)
             }
         }
 
-        private void OnServerConnect(NetworkMessage netMsg)
+        public override void OnServerConnect(NetworkConnection conn)
         {
+            base.OnServerConnect(conn);
+
             Debug.LogWarning("Client Connected");
-            var conn = _connections.Find(c => c.ConnectionId == netMsg.conn.connectionId); 
-            if(conn == null)
+            var uconn = _connections.Find(c => c.ConnectionId == conn.connectionId);
+            if (uconn == null)
             {
                 _connections.Add(new UnityNetworkConnection()
                 {
-                    Connection = netMsg.conn,
-                    ConnectionId = netMsg.conn.connectionId,
+                    Connection = conn,
+                    ConnectionId = conn.connectionId,
                     LobbyId = PlayFabMultiplayerAgentAPI.SessionConfig.SessionId
                 });
             }
         }
 
-        private void OnServerError(NetworkMessage netMsg)
+        public override void OnServerError(NetworkConnection conn, int errorCode)
         {
+            base.OnServerError(conn, errorCode);
+
             try
             {
-                var error = netMsg.ReadMessage<ErrorMessage>();
-                if (error.value != 0)
+                if (errorCode != 0)
```

Comment:
Not sure about this check

Reply:
yeah not sure either, can't find a relevant check in the documentation. Can we remove it completely? Also, now that we don't do ReadMessage I don't think we need the try/catch

---
In reply to: [417784333](https://github.com/PlayFab/gsdkSamples/pull/18#discussion_r417784333) [](ancestors = 417784333)

Reply:
Changed the code as you suggested.